### PR TITLE
support test golang encoding data in java

### PR DESCRIPTION
**Pull request by wongoo**
**What this PR does**:
support test golang encoding data in java

### CODE REVIEW

File: output/output.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/apache/dubbo-go-hessian2/output/testfuncs"
```
Comment (AlexStocks) [inline output/output.go:25]: split this import block.

File: output/output.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+)
+
+type outputFunc func() []byte
+
+var (
+	funcName = flag.String("func_name", "", "func name")
+	funcMap  = make(map[string]outputFunc)
```
Comment (AlexStocks) [inline output/output.go:32]: pls set its init length,

funcMap = make(map[string]outputFunc, 8)

File: output/output.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+	defer func() {
+		if err := recover(); err != nil {
+			_, _ = fmt.Fprintln(os.Stderr, "error: ", err)
+			os.Exit(1)
+		}
+	}()
+	if _, err := os.Stdout.Write(f()); err != nil {
+		_, _ = fmt.Fprintln(os.Stderr, "call error: ", err)
+		os.Exit(1)
```
Comment (AlexStocks) [inline output/output.go:61]: using return instead.

File: output/output.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	}
+	os.Exit(0)
```
Comment (AlexStocks) [inline output/output.go:63]: using return instead.

File: output/output.go (hunk 3 above)
Comment (wongoo) [reply output/output.go:61]: @AlexStocks  main func not support return value

**Pull request merged**